### PR TITLE
Update add-on installation to Django 1.8.7

### PR DESCRIPTION
**Pull request by stugots**
This updates the issue-146 branch so that add-ons can be installed again.

The details of the changes, and how to test them, are in Solinea/goldstone-compliance#28.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -141,7 +141,6 @@ docs/site/
 
 # addon modules
 goldstone/static/addons/
-addon-requirements.txt
```
Comment (jxstanford) [inline .gitignore]: not sure we want to remove this.  if someone checks in the addon-requirements.txt for an addon that is not in the puller's env, they will end up with unnecessary requirements installed.  In the past, the `install_addon/remove_addon` scripts have managed the contents of this file.  It's existence is guaranteed (even if empty) in the docker deployments to ensure that tox has all possible requirements files.

File: addon_fabfile.py [hunk 2]
```diff
@@ -606,13 +607,9 @@ def install_addon(name,
...
+                    # Do a migration to add the add-on's models. (This can't be
                     # done before INSTALLED_APPS is updated.)
-                    error = "doing a syncdb."
-                    _django_manage("makemigrations --noinput " + name,
-                                   proj_settings=settings,
-                                   install_dir=install_dir)
-
+                    error = "doing a migrate."
                     _django_manage("migrate --noinput",
                                    proj_settings=settings,
```
Comment (jxstanford) [inline addon_fabfile.py]: also need to do a `manage.py loaddata` call to address any fixtures that might be present.

File: .gitignore (hunk 1 above)
Comment (stugots) [inline .gitignore]: This file has been moved to the add-ons application directory, and no longer exists here.  And its content has changed.  See below.

File: docker/Dockerfiles/goldstone-app-e/docker_entrypoint.sh [hunk 3]
```diff
@@ -40,7 +40,6 @@ fi
 
 # gather up the static files at container start if this is a dev environment
 if [[ $GS_DEV_ENV == "true" ]] ; then
-    touch addon-requirements.txt
```
Comment (jxstanford) [inline docker/Dockerfiles/goldstone-app-e/docker_entrypoint.sh]: see comment above

File: docker/Dockerfiles/goldstone-app/docker_entrypoint.sh [hunk 4]
```diff
@@ -44,7 +44,6 @@ python manage.py loaddata  # Apply database fixtures
 
 # gather up the static files at container start if this is a dev environment
 if [[ $GS_DEV_ENV == "true" ]] ; then
-    touch addon-requirements.txt
```
Comment (jxstanford) [inline docker/Dockerfiles/goldstone-app/docker_entrypoint.sh]: see comment above

File: tox.ini [hunk 5]
```diff
@@ -20,7 +20,6 @@ skipsdist = True
 [testenv]
 setenv = VIRTUAL_ENV={envdir}
 deps = -r{toxinidir}/requirements.txt
-       -r{toxinidir}/addon-requirements.txt
```
Comment (jxstanford) [inline tox.ini]: see comment above

File: addon_fabfile.py (hunk 2 above)
Comment (stugots) [inline addon_fabfile.py]: AFAICT, the loaddata command requires at least one argument, which is a fixture file.  I can't find any reference to a, "load all the fixture files that exist in the directory hierarchy," command.

It's clear that the Django mothership would prefer that everyone to use "data migrations" for automatically loading arbitrary initial data.  See https://docs.djangoproject.com/en/1.8/topics/migrations/#data-migrations

File: tox.ini (hunk 5 above)
Comment (stugots) [inline tox.ini]: Bug acknowledged, working on a fix.

**Comment by stugots** [comment]
@jxstanford:

I believe the bug you found is now fixed.  Please review again at your convenience...

**Comment by jxstanford** [comment]
:+1:

**Pull request merged**